lib/reqTemplate.js

```diff
@@ -159,6 +162,28 @@ function _createTemplateResolvers(origSpec, subspec, reqPart, path) {
     }
 }
 
+function calculateHash(context) {
+    var req = context.request;
+    // Ensure consistent hashing
+    req.headers = req.headers || {};
+    req.body = req.body || {};
+    req.query = req.query || {};
+    req.params = req.params || {};
+
+    // Need to remove x-request-id header if it's present
+    var reqId = req.headers && req.headers['x-request-id'];
```

Comment:
Not necessary to check for `req.headers` now that we ensure the property exists.